Doc/library/socket.rst

```diff
@@ -909,6 +909,9 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    This is a Python type object that represents the socket object type. It is the
    same as ``type(socket(...))``.
 
+   .. deprecated:: next
+      If type annotations are required, use :class:`~socket.socket` instead.
```

Comment:
```suggestion
      If type annotations are required, use :class:`socket.socket` instead.
```

For type annotations, it's better to use something like `def foo(s: socket.socket)` rather than `def foo(s: socket)` since it could be confused with the `socket` module itself IMO.